collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ActivityGeoDataRequester.java

```diff
@@ -0,0 +1,108 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.activities.GeoPointActivity;
+import org.odk.collect.android.activities.GeoPointMapActivity;
+import org.odk.collect.android.activities.GeoPolyActivity;
+import org.odk.collect.android.listeners.PermissionListener;
+import org.odk.collect.android.utilities.ApplicationConstants;
+import org.odk.collect.android.utilities.PermissionUtils;
+import org.odk.collect.android.utilities.WidgetAppearanceUtils;
+
+public class ActivityGeoDataRequester {
+    public static final String LOCATION = "gp";
+    public static final String ACCURACY_THRESHOLD = "accuracyThreshold";
+    public static final String READ_ONLY = "readOnly";
+    public static final String DRAGGABLE_ONLY = "draggable";
+
+    private final PermissionUtils permissionUtils;
+
+    public ActivityGeoDataRequester() {
```

Comment:
I don't think we should have this constructor. The `WidgetFactory` can construct a `PermissionUtils` object to pass in.